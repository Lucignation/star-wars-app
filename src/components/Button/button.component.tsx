import { FC } from 'react';

import styles from './button.module.scss';

type props = {
  btnText: string;
};
const Button: FC<props> = ({ btnText }) => {
  return <button className={styles.btn}>{btnText}</button>;
};

export default Button;
