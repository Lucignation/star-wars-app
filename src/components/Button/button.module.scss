.btn {
  background: #0a74dc;
  border-radius: 6px;
  border-color: transparent;
  color: #fff;
  width: 100%;
  padding: 12px;

  font-family: 'Inter', sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  /* identical to box height, or 150% */
  cursor: pointer;

  text-align: center;
  letter-spacing: 0.1px;

  /* White */

  color: #ffffff;
}
