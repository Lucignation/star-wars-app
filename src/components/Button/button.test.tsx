import Button from './button.component';
import renderer from 'react-test-renderer';

const mockProps = {
  btnText: 'Log in',
};

describe('Button', () => {
  it('renders a snapshot of <Button /> when passed into a component', () => {
    const button = renderer.create(<Button {...mockProps} />).toJSON();

    expect(button).toMatchSnapshot();
  });
});
