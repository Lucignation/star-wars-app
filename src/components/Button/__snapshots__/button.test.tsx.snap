// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a snapshot of <Button /> when passed into a component 1`] = `
<button
  className="btn"
>
  Log in
</button>
`;
